### PR TITLE
issue_440: Convert Pytest-CVP Bad SysLog test case to Vane

### PR DESCRIPTION
**Pull request by rahulmali2512**
# Please include a summary of the changes

Describe the solution you'd like
PyTest-CVP Bad SysLog test case should have the following:

Use Vane test case styling
Use Master definition file to simplify input
Add test case steps to the test case
Test case should be skipped if SysLog is not configured
Test should be added to top-level directory nrfu_tests
Use markers nrfu_test, system
Use Test ID: NRFU6.1

# Any specific logic/part of code you need extra attention on
   From the output of  **show logging last <daysOfLogs> days**,  verified the key words in syslog events that are generally considered to be bad

# Include the Issue number and link
https://github.com/aristanetworks/vane/issues/440

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU testcases)

# Effort required on reviewers end

- - [ ] Easy
- - [x] Medium
- - [ ] Hard 

# How Has This Been Tested?
Unit testcases:

 - Tested and verified no bad syslog messages are found: Passed
[report_pass_with_test_def_1.docx](https://github.com/aristanetworks/vane/files/12684307/report_pass_with_test_def_1.docx)
[report_pass_with_j2.docx](https://github.com/aristanetworks/vane/files/12684315/report_pass_with_j2.docx)

- Tested and verified bad syslog messages are found: Failed
[report_fail_1.docx](https://github.com/aristanetworks/vane/files/12684322/report_fail_1.docx)

- Tested and verified logging details are not found.: Assert
[report_Assert_1.docx](https://github.com/aristanetworks/vane/files/12684324/report_Assert_1.docx)

- Tested and verified syslog is not configured: Skipped
[report_Skipped_1.docx](https://github.com/aristanetworks/vane/files/12684338/report_Skipped_1.docx)

- html Reports
[issue_30_html_reports.zip](https://github.com/aristanetworks/vane/files/12684347/issue_30_html_reports.zip)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

- Assert reports are generated using hardcoded output.

### CODE REVIEW

File: nrfu_tests/master_def.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+global_dut_filter: # This stanza is optional. If it is not provided by users than all test cases should be run on each dut.
```
Comment (hsisalgslab) [inline nrfu_tests/master_def.yaml:1]: as discussed in DSU, lets not add master def for now

File: nrfu_tests/test_definition.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_bad_syslog_events
+    description: Testcase for verification bad syslog event messages.
+    test_id: NRFU6.1
+    show_cmd: null # Forming show command in test case.
+    expected_output: null # Forming expected output in test case.
+    test_criteria: No bad syslog messages should found. # Setting test case’s pass / fail criteria for reporting
+    report_style: modern # Setting report_style as modern If report type is set to modern, then it will update steps, assumptions, and external systems in the docx report.
+    days_of_logs: 7
```
Comment (hsisalgslab) [inline nrfu_tests/test_definition.yaml:10]: can we confirm ? this can be taken from master_def.yaml

File: nrfu_tests/test_bad_syslog_events.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                    or "not_available" in event.lower()
+                    or "unavailable" in event.lower()
+                    or "interrupt" in event.lower()
+                ):
+                    bad_syslog += f"{event}\n"
+
+            # Collecting actual and expected output.
+            tops.actual_output.update({"bad_syslog_event_found": len(bad_syslog) > 0})
+
+            tops.expected_output.update({"bad_syslog_event_found": False})
```
Comment (hsisalgslab) [inline nrfu_tests/test_bad_syslog_events.py:77]: image
it should be bad_syslog_event_found: true 
discuss if you have any confusion on this

File: nrfu_tests/test_definition.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_bad_syslog_events
```
Comment (hsisalgslab) [inline nrfu_tests/test_definition.yaml:3]:
```suggestion
  - name: test_system_hardware_bad_syslog_event
```
can we update update function like this ?

same for .py file name

File: nrfu_tests/test_bad_syslog_events.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+                if (
+                    "error" in event.lower()
+                    or "not_available" in event.lower()
+                    or "unavailable" in event.lower()
+                    or "interrupt" in event.lower()
+                ):
+                    bad_syslog += f"{event}\n"
+
+            # Collecting actual and expected output.
+            tops.actual_output.update({"bad_syslog_event_found": len(bad_syslog) > 0})
```
Comment (hsisalgslab) [inline nrfu_tests/test_bad_syslog_events.py:75]:
```suggestion
            tops.actual_output.update({"bad_syslog_events_found": len(bad_syslog) > 0})
```

File: nrfu_tests/test_bad_syslog_events.py (hunk 3 above)
Comment (hsisalgslab) [inline nrfu_tests/test_bad_syslog_events.py:77]: image

can we form this starting of the test case so  ?

**Review by hsisalgslab — APPROVED** [review]

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

File: nrfu_tests/master_def.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
...
+  criteria: names
+  filter:
+    - BLFE1
+testcase_data:
+  NRFU6.1:
+    days_of_logs: 7
```
Comment (vishwajeetgslab) [inline nrfu_tests/master_def.yaml:7]: Add inline comment

File: nrfu_tests/test_bad_syslog_events.py [hunk 7]
```diff
@@ -0,0 +1,100 @@
+# Copyright (c) 2023 Arista Networks, Inc.  All rights reserved.
+# Arista Networks, Inc. Confidential and Proprietary.
+
+"""
+Test cases for verification of bad syslog events messages
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_bad_syslog_events.py:5]:
```suggestion
Test cases for verification of bad syslog event messages
```

File: nrfu_tests/test_definition.yaml (hunk 4 above)
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml:3]: format properly.

File: nrfu_tests/test_definition.yaml.j2 [hunk 8]
```diff
@@ -0,0 +1,14 @@
+- name: nrfu_tests
+  testcases:
+    - name: test_bad_syslog_events
+      description: Testcase for verification bad syslog event messages.
+      test_id: NRFU6.1
+      {% set master_data = testcase_data %}
+      {% set log_data = master_data["NRFU6.1"] %}
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml.j2:7]: No need to add these two variables. you can directed add testcase_data["NRFU6.1"] in front of days_of_logs

File: nrfu_tests/test_bad_syslog_events.py [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+                ):
+                    bad_syslog += f"{event}\n"
+
+            # Collecting actual output.
+            tops.actual_output.update({"bad_syslog_events_not_found": not bool(bad_syslog)})
+
+            # Forming output message if test result is fail.
+            if tops.actual_output != tops.expected_output:
+                tops.output_msg = (
+                    f"Following bad syslog events are found on the device: \n{bad_syslog}"
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_bad_syslog_events.py:86]:
```suggestion
                    f"\nFollowing bad syslog events are found on the device: \n{bad_syslog}"
```

File: nrfu_tests/master_def.yaml [hunk 10]
```diff
@@ -1,4 +1,7 @@
-global_dut_filter: # This stanza is optional. If it is not provided by users then all test cases should be run on each dut.
+global_dut_filter: # This stanza is optional. If it is not provided by users than all test cases should be run on each dut.
```
Comment (vishwajeetgslab) [inline nrfu_tests/master_def.yaml:1]: Why we are changing this

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

**Review by vishwajeetgslab — APPROVED** [review]

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**